### PR TITLE
Add bolt db

### LINKED ISSUE #2: Add bolt db
https://github.com/etcd-io/bbolt

### PR DESCRIPTION
**Pull request by MakMuftic**
### Changes

Added **boltdb** and setup code structure as follows (here this is implemented only for Node, but pattern is going to be the same):
- `model` - define structure that represents one entry type in DB and corresponding bucket
- `repository` - hold DB instance and expose a set of action for specific entry type (_Save_, _FindByID_, etc.)
- `controller` - define handlers for a set of endpoints

Restructured the rest of the code based on new changes. Save node inside the register handler and add tests for the handler.

### Issues

closes #2 

### TODO
- Restructure code inside CLI **start** command (_extract database and server initialization to different function_). This will be done when closing issue #8 

- Finish implementing `register handler`. Currently, there is no check if the node is whitelisted. This will be separate PR when closing #7.

### CODE REVIEW

File: internal/controlers/base.go [hunk 1]
```diff
@@ -20,7 +33,7 @@ type RegisterResponse struct {
 	Token string `json:"token"`
 }
 
-func RegisterHandler(w http.ResponseWriter, r *http.Request) {
+func (c BaseController) RegisterHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (mpetrunic) [inline internal/controlers/base.go]: you should probably move this to file register.go inside controllers directory

File: internal/models/node.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package models
+
+type Node struct {
+	ID            string `storm:"id"`
+	ConfigHash    string
+	NodeUrl       string
+	PayoutAddress string `storm:"index"`
```
Comment (mpetrunic) [inline internal/models/node.go]: why do you need index here?

File: internal/controlers/base.go [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
 	"encoding/json"
 	"errors"
 	"github.com/NodeFactoryIo/vedran/internal/auth"
+	"github.com/NodeFactoryIo/vedran/internal/models"
 	"github.com/NodeFactoryIo/vedran/pkg/util"
 	"log"
 	"net/http"
 )
 
+type BaseController struct {
```
Comment (mpetrunic) [inline internal/controlers/base.go]: ApiController might be better name

File: internal/models/node.go (hunk 2 above)
Comment (MakMuftic) [reply internal/models/node.go]: Left from some testing, cleaned :+1:

**Review by mpetrunic — APPROVED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Pull request merged**